python/ray/_private/worker.py

```diff
@@ -516,6 +518,24 @@ def current_task_id(self):
     def current_node_id(self):
         return self.core_worker.get_current_node_id()
 
+    def get_driver_node_id(self, timeout=None) -> ray.NodeID:
+        """
+        Get the driver node id via Job Info. Populates the cached driver node id if
+        found. The fetching can be slow if there are too many jobs.
```

Comment:
Actually fetching can be slow if there are so many workers started at the same time because GCS main thread is overloaded. I've seen it is delayed up to 10~30 seconds.

Ideally, we should not query GCS when a new worker starts, but we are doing this already (each worker pings GCS like 5~6 times when a new worker starts). We can fix this holistically when we address large scale cluster problems.

Reply:
added to comment as TODO